File: app/src/main/java/de/xikolo/events/DownloadStartedEvent.java
package de.xikolo.events;

import de.xikolo.events.base.Event;
import de.xikolo.utils.DownloadUtil;

public class DownloadStartedEvent extends Event {

    public DownloadUtil.AssetDownload download;

    public DownloadStartedEvent(DownloadUtil.AssetDownload download) {
        super(DownloadCompletedEvent.class.getSimpleName() + ": filePath = " + download.getFilePath() + ", type = " + download.getAssetType().toString());
        this.download = download;
    }
}

File: app/src/main/java/de/xikolo/events/DownloadStartedEvent.kt
package de.xikolo.events

import de.xikolo.events.base.Event
import de.xikolo.models.DownloadAsset

class DownloadStartedEvent(val downloadAsset: DownloadAsset) : Event(
    "${DownloadStartedEvent::class.java.simpleName}: filePath = ${downloadAsset.filePath}, type = ${downloadAsset.title}"
)
